File: app/src/main/java/ru/ls/donkitchen/db/table/Receipt.java
package ru.ls.donkitchen.db.table;

import android.provider.BaseColumns;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

/**
 * @author Lord (Kuleshov M.V.)
 * @since 11.11.15
 */
@DatabaseTable(tableName = "receipt")
public class Receipt implements BaseColumns {
	public static final String NAME = "name";
	public static final String INGREDIENTS = "ingredients";
	public static final String CATEGORY_ID = "category_id";
	public static final String RECEIPT = "receipt";
	public static final String IMAGE_LINK = "image_link";
	public static final String VIEWS_COUNT = "views_count";

	@DatabaseField(id = true, canBeNull = false, columnName = _ID)
	public int id;
	@DatabaseField(canBeNull = false, foreign = true, columnName = CATEGORY_ID, foreignColumnName = _ID, foreignAutoRefresh = true)
	public Category category;
	@DatabaseField(canBeNull = false, columnName = NAME)
	public String name;
	@DatabaseField(canBeNull = false, columnName = IMAGE_LINK)
	public String imageLink;
	@DatabaseField(columnName = INGREDIENTS)
	public String ingredients;
	@DatabaseField(canBeNull = false, columnName = RECEIPT)
	public String receipt;
	@DatabaseField(columnName = VIEWS_COUNT)
	public int viewsCount;

	@Override
	public String toString() {
		return "Receipt{" +
				"id=" + id +
				", category=" + category +
				", name='" + name + '\'' +
				", imageLink='" + imageLink + '\'' +
				", ingredients='" + ingredients + '\'' +
				", receipt='" + receipt + '\'' +
				", viewsCount=" + viewsCount +
				'}';
	}
}

File: data/src/main/kotlin/ru/ls/donkitchen/data/db/table/Receipt.kt
package ru.ls.donkitchen.data.db.table

import android.provider.BaseColumns
import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable
import ru.ls.donkitchen.data.db.table.Category

/**
 *
 *
 * @author Lord (Kuleshov M.V.)
 * @since 16.10.16
 */
@DatabaseTable(tableName = "receipt")
class Receipt : BaseColumns {
    companion object {
        const val NAME = "name"
        const val INGREDIENTS = "ingredients"
        const val CATEGORY_ID = "category_id"
        const val RECEIPT = "receipt"
        const val IMAGE_LINK = "image_link"
        const val VIEWS_COUNT = "views_count"
        const val RATING = "rating"
    }

    @DatabaseField(id = true, canBeNull = false, columnName = BaseColumns._ID)
    var id: Int = 0
    @DatabaseField(canBeNull = false, foreign = true, columnName = CATEGORY_ID, foreignColumnName = BaseColumns._ID, foreignAutoRefresh = true)
    var category: Category? = null
    @DatabaseField(canBeNull = false, columnName = NAME)
    var name: String? = null
    @DatabaseField(canBeNull = false, columnName = IMAGE_LINK)
    var imageLink: String? = null
    @DatabaseField(columnName = INGREDIENTS)
    var ingredients: String? = null
    @DatabaseField(canBeNull = false, columnName = RECEIPT)
    var receipt: String? = null
    @DatabaseField(columnName = VIEWS_COUNT)
    var viewsCount: Int = 0
    @DatabaseField(columnName = RATING)
    var rating: Int = 0

    override fun toString(): String {
        return "Receipt(id=$id, category=$category, name=$name, imageLink=$imageLink, ingredients=$ingredients, receipt=$receipt, viewsCount=$viewsCount, rating=$rating)"
    }
}
